src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -113,6 +120,11 @@ internal class InstanceDiscoveryManager : IInstanceDiscoveryManager
             Uri authorityUri = authorityInfo.CanonicalAuthority;
             string environment = authorityInfo.Host;
 
+            if (requestContext.ServiceBundle.Config.UseManagedIdentity)
+            {
+                return DefaultInstanceDiscoveryMetadata;
```

Comment:
I am not sure this will work for non-public clouds. Instead of this, use `CreateEntryForSingleAuthority(authorityInfo.CannonicalAuthority)`